packages/jellyfish-api-core/__tests__/category/masternode/setGov.oracles.splits.test.ts

```diff
@@ -0,0 +1,808 @@
+      tokenA: 'TSLA',
+      tokenB: 'DFI'
+    })
+    await testing.generate(1)
+
+    await testing.token.mint({ amount: 2000, symbol: 'TSLA' })
+    await testing.generate(1)
+
+    await testing.poolpair.add({
+      a: { symbol: 'TSLA', amount: 1 },
+      b: { symbol: 'DFI', amount: 1 }
+    })
+    await testing.generate(1)
+
+    const ppTokenID = Object.keys(await testing.rpc.token.getToken('TSLA-DFI'))[0]
+
+    await testing.rpc.masternode.setGov({
+      ATTRIBUTES: {
+        [`v0/poolpairs/${ppTokenID}/token_a_fee_pct`]: '0.01',
+        [`v0/poolpairs/${ppTokenID}/token_b_fee_pct`]: '0.03'
+      }
+    })
+    await testing.generate(1)
+
+    await testing.rpc.masternode.setGov({ LP_SPLITS: { [Number(ppTokenID)]: 1 } })
+    await testing.generate(1)
+
+    await testing.rpc.masternode.setGov({ LP_LOAN_TOKEN_SPLITS: { [Number(ppTokenID)]: 1 } })
+    await testing.generate(1)
+
+    const tslaDfiInfo = await testing.rpc.token.getToken('TSLA-DFI')
+    tslaDfiPairID = Object.keys(tslaDfiInfo)[0]
+  }
+
+  async function liquidatedVaultSetup (): Promise<void> {
+    await activeVaultSetup()
+
+    // Going to liquidate the vaults by price increase of the loan tokens
+    await testing.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), {
+      prices: [{
+        tokenAmount: '1.49999999@TSLA',
+        currency: 'USD'
+      }]
+    })
+    await testing.container.waitForActivePrice('TSLA/USD', '1.49999999')
+
+    const auction = await testing.rpc.vault.listAuctions()
+    expect(auction.length).toStrictEqual(1)
+    expect(auction[0].state).toStrictEqual('inLiquidation')
+
+    await testing.token.mint({ amount: 0.1, symbol: 'TSLA' })
+    await testing.generate(1)
+
+    await testing.rpc.account.sendTokensToAddress({}, { [collateralAddress]: ['1.1@TSLA'] })
+    await testing.generate(1)
+
+    await testing.rpc.vault.placeAuctionBid({
+      vaultId,
+      index: 0,
+      from: collateralAddress,
+      amount: '1.1@TSLA'
+    })
+    await testing.generate(1)
+  }
+
+  it('should split correctly for single setGOV with valid block number', async () => {
+    const splitBlock1 = await testing.rpc.blockchain.getBlockCount() + 2 // Must be at least 2
+
+    {
+      // Split the oracles
+      const txId = await testing.rpc.masternode.setGov({
+        ATTRIBUTES: {
+          [`v0/oracles/splits/${splitBlock1}`]: `${tslaID}/2`
+        }
+      })
+      await testing.generate(1)
+
+      expect(typeof txId).toStrictEqual('string')
+      expect(txId.length).toStrictEqual(64)
+    }
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${tslaID}`]).toStrictEqual('true') // Always lock token after splitting
+      expect(attributes.ATTRIBUTES[`v0/oracles/splits/${splitBlock1}`]).toStrictEqual('1/2')
+    }
+    await testing.generate(1) // Oracle splitting always needs 2 blocks to be executed
```

Comment:
```suggestion
    await testing.generate(1) 
```
no. split happens at a single designated block `splitBlock1`

Reply:
Can you write a test on this?